Prototype branch for storage config change

andrewbaptist opened the pull request:
The goal is to strip pieces out of this branch into separate PRs

Commit:
storage: create a protobuf to hold StoreSpecs

This commit converts the StoreSpecs to a protobuf structure and moves it
into the storagepb directory. This is now part of the StorageConfig
structure.

Epic: none

Release note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/139953)
<!-- Reviewable:end -->

Commit:
storage: move ExternalIODir into StorageConfig

Previously the ExternalIODir was stored as a variable inside the
BaseConfig struct and updated based on the StoreSpecs if it wasn't
explicitly set. This commit moves it into the StorageConfig and removes
it from the BaseConfig.

Epic: CRDB-41111

Release note: None

Commit:
storage: persist and load the configuration

TODO
